pkg/odo/cli/component/list.go

```diff
@@ -104,57 +147,126 @@ func (lo *ListOptions) Run() (err error) {
 		}
 		return nil
 	}
-	var components component.ComponentList
 
-	if lo.allAppsFlag {
-		// retrieve list of application
-		apps, err := application.List(lo.Client)
+	// non --path workflow below
+	// read the code like
+	// -> experimental
+	//	or	|-> --all-apps
+	//		|-> the current app
+	// -> non-experimental
+	//	or	|-> --all-apps
+	//		|-> the current app
+
+	// experimental workflow
+
+	if lo.isExperimentalMode && util.CheckPathExists(lo.devfilePath) {
+
+		var deploymentList *appsv1.DeploymentList
+		var err error
+
+		// TODO: wrap this into a component list for docker support
+		if lo.allAppsFlag {
+			deploymentList, err = lo.KClient.ListAllDeployments()
```

Comment:
should we only list components created by `odo`? This is listing deployments not created by `odo`:
```
Maysuns-MacBook-Pro:nodejs-ex maysun$ odo list --all-apps
Experimental mode is enabled, use at your own risk

Devfile Components: 
APP     NAME                 PROJECT     TYPE                STATE
        che                  che                             Pushed
        che-operator         che                             Pushed
        devfile-registry     che                             Pushed
        keycloak             che                             Pushed
        plugin-registry      che                             Pushed
        postgres             che                             Pushed
app     somenode             che         nodejs              Pushed
app     testm2               che         java-springboot     Pushed

Openshift Components: 
APP     NAME       PROJECT     TYPE       SOURCETYPE     STATE
app     mynode     che         nodejs     local          Pushed
```

Reply:
ack

Reply:
done